File: ios-native/KPKNFit/KPKNFit/Data/Models/EnergyModels.swift
import Foundation

public enum EnergyConfidence: String, Codable {
    case HIGH
    case MEDIUM
    case LOW
}

public enum EnergyEstimateSource: String, Codable {
    case PLANNED
    case LIVE
    case FINAL
}

public enum DailyEnergyStatus: String, Codable {
    case DEFICIT
    case MAINTENANCE
    case SURPLUS
}

public struct CalorieRange: Codable {
    public let low: Int
    public let mid: Int
    public let high: Int

    public init(low: Int = 0, mid: Int = 0, high: Int = 0) {
        self.low = low
        self.mid = mid
        self.high = high
    }
}

public struct ExerciseEnergyContribution: Codable {
    public let exerciseId: String
    public let exerciseDbId: String?
    public let exerciseName: String
    public let activeKcal: Int
    public let epocKcal: Int
    public let totalKcal: Int
    public let percentageOfSession: Double
    public let completedSets: Int
    public let totalSets: Int

    public init(
        exerciseId: String = "",
        exerciseDbId: String? = nil,
        exerciseName: String = "",
        activeKcal: Int = 0,
        epocKcal: Int = 0,
        totalKcal: Int = 0,
        percentageOfSession: Double = 0.0,
        completedSets: Int = 0,
        totalSets: Int = 0
    ) {
        self.exerciseId = exerciseId
        self.exerciseDbId = exerciseDbId
        self.exerciseName = exerciseName
        self.activeKcal = activeKcal
        self.epocKcal = epocKcal
        self.totalKcal = totalKcal
        self.percentageOfSession = percentageOfSession
        self.completedSets = completedSets
        self.totalSets = totalSets
    }
}

public struct SessionEnergySummary: Codable {
    public let activeKcal: CalorieRange
    public let epocKcal: CalorieRange
    public let totalKcal: CalorieRange
    public let projectedTotalKcal: Int?
    public let confidence: EnergyConfidence
    public let source: EnergyEstimateSource
    public let methodVersion: String
    public let exerciseContributions: [ExerciseEnergyContribution]
    public let notes: [String]

    public init(
        activeKcal: CalorieRange = CalorieRange(),
        epocKcal: CalorieRange = CalorieRange(),
        totalKcal: CalorieRange = CalorieRange(),
        projectedTotalKcal: Int? = nil,
        confidence: EnergyConfidence = .LOW,
        source: EnergyEstimateSource = .PLANNED,
        methodVersion: String = "auge-energy-v1",
        exerciseContributions: [ExerciseEnergyContribution] = [],
        notes: [String] = []
    ) {
        self.activeKcal = activeKcal
        self.epocKcal = epocKcal
        self.totalKcal = totalKcal
        self.projectedTotalKcal = projectedTotalKcal
        self.confidence = confidence
        self.source = source
        self.methodVersion = methodVersion
        self.exerciseContributions = exerciseContributions
        self.notes = notes
    }
}

public struct DailyEnergyBalance: Codable {
    public let consumedKcal: Int
    public let trainingBurnKcal: Int
    public let netKcal: Int
    public let targetKcal: Int
    public let deltaFromTarget: Int
    public let status: DailyEnergyStatus

    public init(
        consumedKcal: Int = 0,
        trainingBurnKcal: Int = 0,
        netKcal: Int = 0,
        targetKcal: Int = 0,
        deltaFromTarget: Int = 0,
        status: DailyEnergyStatus = .MAINTENANCE
    ) {
        self.consumedKcal = consumedKcal
        self.trainingBurnKcal = trainingBurnKcal
        self.netKcal = netKcal
        self.targetKcal = targetKcal
        self.deltaFromTarget = deltaFromTarget
        self.status = status
    }
}

File: android-native/app/src/main/java/com/example/kpkn/data/models/EnergyModels.kt
package com.example.kpkn.data.models

import kotlinx.serialization.Serializable

@Serializable
enum class EnergyConfidence { HIGH, MEDIUM, LOW }

@Serializable
enum class EnergyEstimateSource { PLANNED, LIVE, FINAL }

@Serializable
enum class DailyEnergyStatus { DEFICIT, MAINTENANCE, SURPLUS }

@Serializable
data class CalorieRange(
    val low: Int = 0,
    val mid: Int = 0,
    val high: Int = 0,
)

@Serializable
data class ExerciseEnergyContribution(
    val exerciseId: String = "",
    val exerciseDbId: String? = null,
    val exerciseName: String = "",
    val activeKcal: Int = 0,
    val epocKcal: Int = 0,
    val totalKcal: Int = 0,
    val percentageOfSession: Double = 0.0,
    val completedSets: Int = 0,
    val totalSets: Int = 0,
)

@Serializable
data class SessionEnergySummary(
    val activeKcal: CalorieRange = CalorieRange(),
    val epocKcal: CalorieRange = CalorieRange(),
    val totalKcal: CalorieRange = CalorieRange(),
    val projectedTotalKcal: Int? = null,
    val confidence: EnergyConfidence = EnergyConfidence.LOW,
    val source: EnergyEstimateSource = EnergyEstimateSource.PLANNED,
    val methodVersion: String = "auge-energy-v2",
    val exerciseContributions: List<ExerciseEnergyContribution> = emptyList(),
    val notes: List<String> = emptyList(),
)

@Serializable
data class DailyEnergyBalance(
    val consumedKcal: Int = 0,
    val trainingBurnKcal: Int = 0,
    val netKcal: Int = 0,
    val targetKcal: Int = 0,
    val deltaFromTarget: Int = 0,
    val status: DailyEnergyStatus = DailyEnergyStatus.MAINTENANCE,
)
